src/world/partner/bombette.c

```diff
@@ -240,80 +246,74 @@ ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall) {
-            npc->pos.x += (npc->moveToPos.x -  npc->pos.x) / npc->duration;
-            npc->pos.y += (npc->moveToPos.y - npc->pos.y) /  npc->duration;
-            npc->pos.z += (npc->moveToPos.z - npc->pos.z) /  npc->duration;
+            npc->pos.x += (npc->moveToPos.x - npc->pos.x) / npc->duration;
+            npc->pos.y += (npc->moveToPos.y - npc->pos.y) / npc->duration;
+            npc->pos.z += (npc->moveToPos.z - npc->pos.z) / npc->duration;
             npc->duration--;
-
             if (npc->duration != 0) {
                 break;
             }
-
             if (evt->functionTemp[3] < playerStatus->inputEnabledCounter) {
                 disable_npc_blur(npc);
-                evt->functionTemp[0] = 7;
-            } else {
-                npc->pos.x = npc->moveToPos.x;
-                npc->pos.y = npc->moveToPos.y;
-                npc->pos.z = npc->moveToPos.z;
-                disable_npc_blur(npc);
-                suggest_player_anim_clearUnkFlag(ANIM_Mario_6000C);
-                npc->yaw = playerStatus->targetYaw;
-                npc->currentAnim = ANIM_WorldBombette_Walk;
-                evt->functionTemp[0] = 1;
-                evt->functionTemp[1] = 10;
+                temp_f0 = 0;
+                evt->functionTemp[(u8)temp_f0] = 7;
+                break;
             }
+            npc->pos.x = npc->moveToPos.x;
+            npc->pos.y = npc->moveToPos.y;
+            npc->pos.z = npc->moveToPos.z;
+            disable_npc_blur(npc);
+            suggest_player_anim_clearUnkFlag(ANIM_Mario_6000C);
+            npc->yaw = playerStatus->targetYaw;
+            npc->currentAnim = ANIM_WorldBombette_Walk;
+            evt->functionTemp[0] = 1;
+            evt->functionTemp[1] = 10;
         case 1:
-            if (playerStatus->actionState == 0x15 || playerStatus->actionState == 0x16) {
-                evt->functionTemp[1] = 7;
+            if (playerStatus->actionState == ACTION_STATE_HIT_FIRE || playerStatus->actionState == ACTION_STATE_KNOCKBACK) {
+                evt->functionTemp[0] = 7;
                 break;
             }
             npc->pos.y = playerStatus->position.y + playerStatus->colliderHeight;
             npc->yaw = playerStatus->targetYaw;
             if (evt->functionTemp[1] == 1) {
                 suggest_player_anim_clearUnkFlag(ANIM_Mario_60006);
             }
-
             if (evt->functionTemp[1] != 0) {
                 evt->functionTemp[1]--;
                 break;
             }
-
             sfx_play_sound_at_npc(SOUND_80000000, 0, -4);
             D_802BE924 = 1;
             add_vec2D_polar(&npc->pos.x, &npc->pos.z, 0.0f, npc->yaw);
             npc->currentAnim = ANIM_WorldBombette_WalkLit;
             npc->jumpVelocity = 0.0f;
             D_802BE938 = 0;
-            npc->flags = (npc->flags | 0x200) & ~0x100;
+            npc->flags = (npc->flags | 0x200) & (~0x100);
```

Comment:
these should be two separate lines ( flags |= 0x200, flags &= ~0x100)
and they should use the npc flag enum values